lib/adapters/file-adapter.js

```diff
@@ -56,10 +72,28 @@ var Read = Juttle.proc.source.extend({
     start: function() {
         var self = this;
         var stream = this.fetch();
+        var last;
 
         self.parser.parseStream(stream, function(points) {
+            var validPoints = [];
             self.parseTime(points, self.timeField);
-            self.emit(points);
+            _.each(points, function(point) {
+                if (last && last.gt(point.time)) {
+                    // out of order points are dropped
+                    self.trigger('warning', self.runtime_error('RT-OUT-OF-ORDER'), {
+                        proc: 'read file'
+                    });
+                } else {
+                    // timeless points just pass through at this point
+                    if (point.time && point.time.moment) {
+                        if (point.time.gte(self.from) && point.time.lt(self.to)) {
+                            validPoints.push(point);
+                        }
+                    }
+                }
+                last = point.time;
```

Comment:
This should be moved up into the `else` above so that times of OUT-OF-ORDER points don't move `last` backwards.